### DISCUSSION TITLE
Upgrade version 3.2.1 to 3.4.2

### DISCUSSION THREAD

**Discussion by mroupie34** [netbox-community/netbox#11542] [General]
Hello everyone,

I got a problem about this uprade (screen)

![Netbox](https://user-images.githubusercontent.com/107417012/213688770-e2c1a0c7-f590-44f1-94f3-2ad3e86844d1.JPG)

I followed the documentation : https://docs.netbox.dev/en/stable/installation/upgrading/

Vérification : 

PostegreSQL Current Version :
root@MGTVLDCIMNA01:~# psql --version
psql (PostgreSQL) 12.12 (Ubuntu 12.12-0ubuntu0.20.04.1)

![PostegreSQL_V](https://user-images.githubusercontent.com/107417012/213688213-1bec256c-15ea-4522-9dc5-bfe48b33bb74.JPG)

![PostegreSQL](https://user-images.githubusercontent.com/107417012/213687878-bee1605a-7e3c-4074-9f69-8f8a698e5fc7.JPG)

Python Current Version :
root@MGTVLDCIMNA01:~# python3 --version
Python 3.8.10

Redis Current Version :
root@MGTVLDCIMNA01:~# redis-server --version
Redis server v=5.0.7 sha=00000000:0 malloc=jemalloc-5.2.1 bits=64 build=66bd629f924ac924

Django Current Version :
root@MGTVLDCIMNA01:/opt/netbox/netbox# django-admin --version
4.0.4

Upgrade to 3.4.2
```
---------------------------------------------------------------------------
wget https://github.com/netbox-community/netbox/archive/v3.4.2.tar.gz
sudo tar -xzf v3.4.2.tar.gz -C /opt
sudo ln -sfn /opt/netbox-3.4.2/ /opt/netbox
---------------------------------------------------------------------------
sudo cp /opt/netbox-3.2.1/local_requirements.txt /opt/netbox/
sudo cp /opt/netbox-3.2.1/netbox/netbox/configuration.py /opt/netbox/netbox/netbox/
sudo cp /opt/netbox-3.2.1/netbox/netbox/ldap_config.py /opt/netbox/netbox/netbox/
---------------------------------------------------------------------------
sudo cp -pr /opt/netbox-3.2.1/netbox/media/ /opt/netbox/netbox/
---------------------------------------------------------------------------
sudo cp /opt/netbox-3.2.1/gunicorn.py /opt/netbox/
---------------------------------------------------------------------------
sudo /opt/netbox/upgrade.sh
---------------------------------------------------------------------------
sudo pip3 install -r /opt/netbox/requirements.txt
---------------------------------------------------------------------------
sudo systemctl restart netbox netbox-rq
---------------------------------------------------------------------------
```
![Netbox_1](https://user-images.githubusercontent.com/107417012/213686688-79018f9a-257d-4d0f-bd8f-fac2e79682d7.JPG)

What can i do for to change the target version of postegresql ?

Thanks a lot for your feedback
Max

**Comment by mroupie34** [comment]
Hi everyone,

I resolved this problem like this  :)

Upgrade 3.2.1 -> 3.2.6 = OK
Upgrade 3.2.6 -> 3.2.7 = OK
Upgrade 3.2.7 -> 3.3.0 = OK
Upgrade 3.3.0 -> 3.3.10 = OK
Upgrade 3.3.10 -> 3.4.0 = KO  postgreSQL not supported version 10

-> Upgrading PostgreSQL Version on Ubuntu Server : https://gorails.com/guides/upgrading-postgresql-version-on-ubuntu-server

Upgrade 3.3.10 -> 3.4.0 = OK
Upgrade 3.4.0 -> 3.4.2 = OK

---------------------------------------------

/opt
```
---------------------------------------------------------------------------
wget https://github.com/netbox-community/netbox/archive/v3.4.2.tar.gz
sudo tar -xzf v3.4.2.tar.gz -C /opt
sudo ln -sfn /opt/netbox-3.4.2/ /opt/netbox
---------------------------------------------------------------------------
sudo cp /opt/netbox-3.2.1/local_requirements.txt /opt/netbox/
sudo cp /opt/netbox-3.2.1/netbox/netbox/configuration.py /opt/netbox/netbox/netbox/
sudo cp /opt/netbox-3.2.1/netbox/netbox/ldap_config.py /opt/netbox/netbox/netbox/
---------------------------------------------------------------------------
sudo cp -pr /opt/netbox-3.2.1/netbox/media/ /opt/netbox/netbox/
---------------------------------------------------------------------------
sudo cp /opt/netbox-3.2.1/gunicorn.py /opt/netbox/
---------------------------------------------------------------------------
sudo /opt/netbox/upgrade.sh
---------------------------------------------------------------------------
sudo pip3 install -r /opt/netbox/requirements.txt
---------------------------------------------------------------------------
sudo systemctl restart netbox netbox-rq
---------------------------------------------------------------------------
```



Thanks a lot
Max

**Reply by mtinberg** [reply]
I compliment you on your clear documentation of your troubleshooting and resolution process.  Glad you got it working

**Reply by candlerb** [reply]
Yes, well done.

Just one thing would make it clearer: put three backticks (` ``` `) on a line of their own, before and after each block of terminal session you're pasting.  This gives you fixed-format code blocks, instead of it being interpreted as markdown (this is why you get the bold text etc).

You can edit the existing post to add them.

**Reply by mroupie34** [reply]
@mtinberg Thanks a lot !
@candlerb i think it will be good now :)
